### DISCUSSION TITLE
Receive messages

### DISCUSSION THREAD

**Discussion by pacos1** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#24] [Q&A] [answered]
First of all thank you for your work.

I can´t receive messages sent from my phone to the test number.
I am thinking is the routes, I have [Route("myroute")] at the beginning of WhatsAppBusinessCloudAPI.Web.Controllers namespace for all the methods but maybe each method needs different routes. I test different routes though, still not working.

Thanks in advance.

**Comment by gabrieldwight** [comment, marked as answer]
Hi @pacos1, Did you verify the route endpoint for your server that meta sent a verification code to when registering the webhook? Once verified you can subscribe your verified webhook endpoint to the **message service** on the developer portal. 

You can only use one callback endpoint to subscribe to many meta notification services. 

If you are using the test environment then you will need to use the local tunnel software extension on the visual studio to receive the message update.

**Reply by pacos1** [reply]
That´s it. I forgot to subscribe to message service.
Thank you very much.
